fedbiomed/common/channel_manager.py

```diff
@@ -0,0 +1,186 @@
+                or None if no channel exists for this `distant_node_id`
+
+        Raises:
+            FedbiomedNodeToNodeError: failed to query the database
+            FedbiomedNodeToNodeError: more than one entry in database with this node ID
+        """
+        try:
+            entries = self._table.search(
+                self._query.distant_node_id.exists() &
+                (self._query.distant_node_id == distant_node_id)
+            )
+        except Exception as e:
+            errmess = f'{ErrorNumbers.FB630.value}: failed searching the database table "{self._table}" ' \
+                      f'for channel distant_node_id="{distant_node_id}" with error: {e}'
+            logger.error(errmess)
+            raise FedbiomedNodeToNodeError(errmess)
+
+        if len(entries) > 1:
+            errmess = f'{ErrorNumbers.FB630.value}: database table "{self._table}" is inconsistent: ' \
+                      f'found {len(entries)} entries with unique distant_node_id={distant_node_id}'
+            logger.error(errmess)
+            raise FedbiomedNodeToNodeError(errmess)
+        elif len(entries) == 1:
+            element = entries[0]
+            raise_for_version_compatibility(
+                element.get('n2n_channel_version', __default_version__),
+                __n2n_channel_element_version__,
+                f"{ErrorNumbers.FB625.value}: Incompatible versions  (found %s but expected %s) for "
+                f"channel distant_node_id={distant_node_id} in database {self._table}"
+            )
+        else:
+            element = None
+
+        if element:
+            # Need to convert to bytes
+            element['local_key'] = bytes(base64.b64decode(element['local_key']))
+
+        return element
+
+    def add(self, distant_node_id: str, local_key: bytes):
+        """Add a new data entry for this `distant_node_id` in database
+
+        Check that no entry exists yet for `distant_node_id` in the table.
+
+        Args:
+            distant_node_id: unique id used to save a given channel component
+            local_key: private key of local endpoint for this channel
+
+        Raises:
+            FedbiomedNodeToNodeError: failed to insert in database
+        """
+        self._check_existing_entry_in_db(distant_node_id)
+        channel = {
+            'n2n_channel_version': str(__n2n_channel_element_version__),
+            'distant_node_id': distant_node_id,
+            'local_key': str(base64.b64encode(local_key), 'utf-8'),
+        }
+
+        try:
+            self._table.insert(channel)
+        except Exception as e:
+            errmess = f'{ErrorNumbers.FB630.value}: failed adding an entry in table "{self._table}" ' \
+                      f'for distant_node_id={distant_node_id} with error: {e}'
+            logger.error(errmess)
+            raise FedbiomedNodeToNodeError(errmess)
+
+    def remove(self, distant_node_id: str) -> bool:
```

Comment:
Hi @srcansiz you are right it is not used for now.

I proceeded the same as for secagg elements databases: it seemed to me weird not to provide a method for removing an element. This can be used by advanced user to selectively reset a channel's key. This may be used in the future for some function included in the lbrary.

If you prefer, I can remove it.

Reply:
Hi @mvesin,
Thanks, Let's remove it and add it if we need it in the future.